src/Common/src/System/Security/Cryptography/CryptoPool.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    internal static class CryptoPool
+    {
+        internal const int ClearAll = -1;
+
+        private static readonly ArrayPool<byte> s_pool = ArrayPool<byte>.Create();
```

Comment:
This isn't just introducing one separate pool... it's introducing one separate pool per assembly that imports this file, which looks to be ~10.

I question whether this change is really worthwhile.  I read your justification, but I don't see why these libraries are significantly more at risk than every other library we have that uses the shared pool.  Everything is in process, where there's no security boundary.  We're already being a little paranoid and aggressively clearing when we put the arrays back.  As Levi pointed out, the pool created by Create is also less sophisticated and doesn't trim.  It also has a difference performance profile, so any measurements we did in the past around throughput might be impacted.